src/core/dev/allocator.cpp

```diff
@@ -62,7 +64,7 @@ xlio_allocator::xlio_allocator(alloc_t alloc_func, free_t free_func)
     m_size = 0;
     m_memalloc = alloc_func;
     m_memfree = free_func;
-    if (m_memalloc && m_memfree) {
+    if (m_memalloc) {
```

Comment:
why not to check pair `m_memalloc && m_memfree`?

Reply:
Because it's logically possible to provide only allocator and omit free function. XLIO will only allocate memory. On the other hand, if only free function is set for some reason - it will never be used, so we can ignore it.

ioctl() doesn't allow to set only a single function, however, allocator component allows this in case this is changed in the future.